kernel/Kconfig

```diff
@@ -928,4 +928,15 @@ config THREAD_LOCAL_STORAGE
 
 endmenu
 
+menu "Device Options"
+
+config HAS_DYNAMIC_DEVICE_HANDLES
```

Comment:
nit: `HAS` may be changed to something else for clarity (we don't have anything, it's an option to select a requirement), maybe `USE` or `NEEDS` (not sure what if there's a pattern).

Reply:
There are some places using `NEEDS` but the majority of similar cases are using `HAS`.